### PR TITLE
Update .clang-format

### PR DESCRIPTION
**Pull request by 375gnu**
I propose to make this style mandatory for all our new code. It's based on our current style, but a few rules might be found controversial. Please leave comments, raise your concerns.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -1,47 +1,63 @@
...
-AllowAllParametersOfDeclarationOnNextLine: false
-AllowShortFunctionsOnASingleLine: true
-AllowShortIfStatementsOnASingleLine: false
-AllowShortLoopsOnASingleLine: false
-AlwaysBreakBeforeMultilineStrings: false
-AlwaysBreakTemplateDeclarations: true
-BinPackParameters: false
-BreakBeforeBinaryOperators: false
+---
+AccessModifierOffset: '-3'
```
Comment (ajtribick) [inline .clang-format:2]: Presumably this gives us the 1 space indent for the access modifiers. I think we should switch this to -4 and reformat the code.

File: .clang-format [hunk 2]
```diff
@@ -1,47 +1,63 @@
...
+AlignAfterOpenBracket: Align
```
Comment (ajtribick) [inline .clang-format:3]: I wonder if we should switch to `AlwaysBreak` - the hanging indent style ends up with everything far off to the right of the window, which can get a bit annoying.

**Comment by 375gnu** [comment]
@levinli303 what do you think?

File: .clang-format [hunk 3]
```diff
@@ -1,47 +1,63 @@
...
+AlignEscapedNewlines: Left
+AlignOperands: 'true'
+AlignTrailingComments: 'true'
+AllowAllArgumentsOnNextLine: 'false'
+AllowAllConstructorInitializersOnNextLine: 'false'
+AllowAllParametersOfDeclarationOnNextLine: 'false'
+AllowShortBlocksOnASingleLine: 'false'
+AllowShortCaseLabelsOnASingleLine: 'false'
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: WithoutElse
```
Comment (375gnu) [inline .clang-format:16]: maybe still be better to return to `false`

File: .clang-format (hunk 2 above)
Comment (375gnu) [reply .clang-format:3]: with `AlwaysBreak` it also looks ugly :)

**Review by levinli303 — APPROVED** [review]

**Comment by 375gnu** [comment]
@ajtribick i've updated according with your review, plz check

**Review by ajtribick — APPROVED** [review]
Ok, let's see how this goes...

**Pull request merged**